### PR TITLE
Update Slurm Cloud SQL example to V6

### PR DESCRIPTION
**Pull request by alyssa-sm**
This PR updates the Slurm references to point to the tag 6.3.2 which is connected to the fix for using CloudSQL with V6. Additionally, it also updates the CloudSQL blueprint to use V6.

Here is an example of a job submitted after making these changes.
```
[alyssasm_hpctoolkit_joonix_net@hpccluster-controller ~]$ scontrol show job 4
JobId=4 JobName=wrap
   UserId=alyssasm_hpctoolkit_joonix_net(1027357380) GroupId=alyssasm_hpctoolkit_joonix_net(1027357380) MCS_label=N/A
   Priority=4294901756 Nice=0 Account=(null) QOS=normal
   JobState=COMPLETED Reason=None Dependency=(null)
   Requeue=1 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0
   RunTime=00:00:10 TimeLimit=UNLIMITED TimeMin=N/A
   SubmitTime=2024-01-26T20:40:34 EligibleTime=2024-01-26T20:40:34
   AccrueTime=2024-01-26T20:40:34
   StartTime=2024-01-26T20:42:18 EndTime=2024-01-26T20:42:28 Deadline=N/A
   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2024-01-26T20:40:34 Scheduler=Main
   Partition=compute AllocNode:Sid=hpccluster-controller:3180
   ReqNodeList=(null) ExcNodeList=(null)
   NodeList=hpccluster-comput-0
   BatchHost=hpccluster-comput-0
   NumNodes=1 NumCPUs=2 NumTasks=1 CPUs/Task=1 ReqB:S:C:T=0:0:*:*
   ReqTRES=cpu=1,mem=7752M,node=1,billing=1
   AllocTRES=cpu=2,mem=15504M,node=1,billing=2
   Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*
   MinCPUsNode=1 MinMemoryCPU=7752M MinTmpDiskNode=0
   Features=(null) DelayBoot=00:00:00
   OverSubscribe=NO Contiguous=0 Licenses=(null) Network=(null)
   Command=(null)
   WorkDir=/home/alyssasm_hpctoolkit_joonix_net
   StdErr=/home/alyssasm_hpctoolkit_joonix_net/slurm-4.out
   StdIn=/dev/null
   StdOut=/home/alyssasm_hpctoolkit_joonix_net/slurm-4.out
   Power=
```

### CODE REVIEW

File: tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml [hunk 1]
```diff
@@ -42,32 +42,36 @@ deployment_groups:
       sql_instance_name: slurm-sql8
       tier: "db-f1-micro"
 
+  - id: compute-nodeset
+    source: ./community/modules/compute/schedmd-slurm-gcp-v6-nodeset
```
Comment (nick-stroud) [inline tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml:46]: nit: Remove the `./` from each source line. See L29 for example.

File: tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml
Comment (nick-stroud) [inline tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml:1]: Rename title to "Update Slurm Cloud SQL example to V6"

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf:1]: There will be similar version references in login, partition, and nodeset modules. I think best thing would be to update all of them to keep them in sync.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf
Comment (alyssa-sm) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf:1]: I replaced all instances of the previous version I could find. Did you see more to update?

Comment (nick-stroud) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/controller.tf:1]: silly me. The other modules are just data representations. All the work is happening in the controller.

**Review by nick-stroud — APPROVED** [review]

**Comment by harshthakkar01** [comment]
lgtm. good to go after v6 tests pass.

**Review by harshthakkar01 — APPROVED** [review]

File: tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml (hunk 1 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/hpc-cluster-simple-nfs-sql.yaml]: Removed.

**Pull request merged**